mirrord/layer/tests/common/mod.rs

```diff
@@ -183,20 +183,26 @@ impl LayerConnection {
         app_port: u16,
     ) -> LayerConnection {
         let mut codec = Self::accept_library_connection(listener).await;
-        let msg = match codec.next().await {
-            Some(option) => option.unwrap(),
-            None => {
-                // Python runs in 2 processes, only one of which is the application. The library is
-                // loaded into both so the first connection will not contain the application and
-                // so will not send any of the messages that are generated by the hooks that are
-                // triggered by the app.
-                // So accept the next connection which will be the one by the library that was
-                // loaded to the python process that actually runs the application.
-                codec = Self::accept_library_connection(listener).await;
-                codec.next().await.unwrap().unwrap()
-            }
-        };
-        assert_eq!(msg, ClientMessage::Tcp(LayerTcp::PortSubscribe(app_port)));
+
+        if codec.next().await.is_none() {
```

Comment:
Here you are dropping the next message of the codec, even if it's `Some`. What are the current changes in this file for?
When checking out this file from main into this branch, only the one test that is actually affected by the change in gethostname fails, instead of multiple tests that use this function.

For that test you would need to change this function to optionally receive and answer a `FileRequest(Open(OpenFileRequest { path: "/etc/hostname" ... }))` before the `PortSubscribe`, or maybe instead of making this function also accept that case, create a `get_initialized_connection_with_hostname_and_port` function that is like this one but always expects a hostname `FileRequest` - up to you.

The integration tests work by opening a TCP port for the layer to connect to and then answering as an agent would, simulating the agent for the layer. This means that the integration tests have to anticipate and answer each message that the layer is going to send in the scenario they create. Because of the `gethostname` change, in one of the tests (on mac it's only one, maybe it's different on Linux), the layer sends a `FileRequest` before it sends a `PortSubscribe`, so in that case the integration test has to read that message and then send a response.

Let me know if that explanation wasn't clear, we can talk about it over voice :)